include/deal.II/fe/fe_tools.templates.h

```diff
@@ -421,106 +421,110 @@ namespace FETools
       // to this one, taking into account multiplicities, multiple components in
       // base elements, and other complications
       unsigned int total_index = 0;
-      for (const unsigned int vertex_number :
-           fes.front()->reference_cell().vertex_indices())
-        {
-          unsigned int comp_start = 0;
-          for (unsigned int base = 0; base < fes.size(); ++base)
-            for (unsigned int m = 0; m < multiplicities[base];
-                 ++m,
-                              comp_start +=
-                              fes[base]->n_components() * do_tensor_product)
-              for (unsigned int local_index = 0;
-                   local_index < fes[base]->n_dofs_per_vertex();
-                   ++local_index, ++total_index)
-                {
-                  const unsigned int index_in_base =
-                    (fes[base]->n_dofs_per_vertex() * vertex_number +
-                     local_index);
-
-                  Assert(comp_start + fes[base]->n_components() <=
-                           retval[total_index].size(),
-                         ExcInternalError());
-                  for (unsigned int c = 0; c < fes[base]->n_components(); ++c)
-                    {
-                      Assert(c < fes[base]
-                                   ->get_nonzero_components(index_in_base)
-                                   .size(),
-                             ExcInternalError());
-                      retval[total_index][comp_start + c] =
-                        fes[base]->get_nonzero_components(index_in_base)[c];
-                    }
-                }
-        }
+      if (fes.size() > 0)
```

Comment:
Does this function even make sense if `fes.size()==0`? If not, I would prefer having an assertion at the top.